Plugins/Json/include/Acts/Plugins/Json/SurfaceJsonConverter.hpp

```diff
@@ -21,16 +21,17 @@
 // can not match our naming guidelines.
 namespace Acts {
 
-using SurfaceAndMaterial =
-    std::pair<std::shared_ptr<const Acts::Surface>,
-              std::shared_ptr<const Acts::ISurfaceMaterial>>;
+using SurfaceAndMaterialWithContext =
+    std::tuple<std::shared_ptr<const Acts::Surface>,
+               std::shared_ptr<const Acts::ISurfaceMaterial>,
+               const Acts::GeometryContext*>;
```

Comment:
Hmm. In principle, `GeometryContext` must be copyable. Maybe we should just put it in here by value, and copy it when writing to JSON? Basically, taking pointers to (potentially) stack variables makes me a bit uncomfortable, that's why.

Reply:
I see your point, I was a bit worried about doing the copy for every single surface (since that is a lot of surfaces). But I can change it to do the copy for sure.